### DISCUSSION TITLE
Expand the use of TimeStampedProcessVariable

### DISCUSSION THREAD

**Discussion by shroffk** [ControlSystemStudio/phoebus#2191] [Ideas]
When exporting the selection from applications like alarm views, save restore... it makes sense to set the selection to contain the timestamp.

Context menu action to open the databrowser can use the Instant to then open the plot initialized to a more helpful time range.

e.g. when you are looking at pv in the save restore snapshot or the alarm log or the alarm view...it makes sense to open the databrowser to somewhere around that time range (+/- 30 mins) instead of opening it to now and then having users manually adjust the time range.

**Comment by georgweiss** [comment]
Is support for multiple TimeStampedProcessVariable(s) in the selection implied?

**Comment by shroffk** [comment]
I think we should be able to handle this.
Selection = a list of TimestampedProcessVariable
Context menu actions = The tool handling the selection can adopt appropriate strategies... for the databrowser, the start time of the plot can be the earliest timestamp and the end time can the the latest timestamp associated with the selected TPV

**Reply by georgweiss** [reply]
How does one determine "earliest" and "latest" timestamp? Is there an archiver API for that?

**Reply by shroffk** [reply]
using to the compareTo() method of Instant, each TimestampedProcessVariable has an Instant associated with it

**Reply by georgweiss** [reply]
Still do not understand. We need a time range, a single TimestampedProcessVariable does not define one. 

From a snapshot example at ESS I can see PVs with essentially timestamp zero, guess those records were never processed.
